xdsl/traits.py

```diff
@@ -94,6 +94,75 @@ def verify(self, op: Operation) -> None:
                 )
 
 
+class SingleBlockImplicitTerminator(OpTrait):
+    """
+    Checks and adds if missing the specified terminator to an operation which has
+    single-block region.
+
+    This should be fully compatible with MLIR's Trait:
+    https://mlir.llvm.org/docs/Traits/#single-block-with-implicit-terminator
+    """
+
+    parameters: type[Operation]
+
+    def __init__(self, parameters: type[Operation]):
+        super().__init__(parameters)
+
+    def ensure_terminator(self, op: Operation) -> None:
```

Comment:
I would actually move this outside of this class, what do you think?
So users would write `ensure_terminator(self)`, instead of going through the traits. What do you think?

Reply:
That sounds good.
There might be reuses for `ImplicitAffineTerminator`.